include/zephyr/sys/sflist.h

```diff
@@ -288,14 +283,15 @@ static inline uint8_t sys_sfnode_flags_get(sys_sfnode_t *node)
  * @brief Initialize an sflist node
  *
  * Set an initial flags value for this slist node, which can be a value between
- * 0 and 3. These flags will persist even if the node is moved around
- * within a list, removed, or transplanted to a different slist.
+ * 0 and 3 on 32-bit architectures, or between 0 and 7 on 64-bit architectures.
```

Comment:
Do we need this extra bit? It seems like you're introducing a potential architecture-dependent behavior here. Maybe just always provide two bits and avoid this?